doh.c

```diff
@@ -763,7 +767,17 @@ int main(int argc, char **argv)
     case 'V': /* version */
       show_version();
       break;
-    case 't': /* test mode */
+    case 't':
+      if(strncmp(&argv[0][2], "AAAA", 4) == 0)
+        query_type = DNS_TYPE_AAAA;
+      else if(strncmp(&argv[0][2], "A", 1) == 0)
+        query_type = DNS_TYPE_A;
+      else if(strncmp(&argv[0][2], "CNAME", 5) == 0)
+        query_type = DNS_TYPE_CNAME;
+      else if(strncmp(&argv[0][2], "TXT", 3) == 0)
+        query_type = DNS_TYPE_TXT;
```

Comment:
what if an unknown/unrecognized type is specified? shouldn't it say something or bail or so?

Reply:
I do `help("type not supported");` on line 820 if none of the types is met. Is it what is needed or I got you wrong?

Reply:
here https://github.com/curl/doh/blob/18b00d5f2e12059cad489e9f998cc39f937afdea/doh.c#L820